### PR TITLE
Update Istio and Sail remaining jobs to use OCP 4.16

### PR DESCRIPTION
**Pull request by fjglira**

### CODE REVIEW

File: ci-operator/config/openshift-service-mesh/istio/openshift-service-mesh-istio-master.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ tests:
       RELEASE_IMAGE_LATEST: release:latest
     env:
       BASE_DOMAIN: servicemesh.devcluster.openshift.com
-      HYPERSHIFT_HC_RELEASE_IMAGE: quay.io/openshift-release-dev/ocp-release:4.15.6-multi
+      HYPERSHIFT_HC_RELEASE_IMAGE: quay.io/openshift-release-dev/ocp-release:4.16.3-multi
```
Comment (jwendell) [inline ci-operator/config/openshift-service-mesh/istio/openshift-service-mesh-istio-master.yaml:63]: why not 4.16.5, which seems to be the latest?

Comment (fjglira) [reply ci-operator/config/openshift-service-mesh/istio/openshift-service-mesh-istio-master.yaml]: Done

**Comment by fjglira** [comment]
`pull-ci-openshift-service-mesh-sail-operator-release-3.0-tp1-e2e-ocp-4-16` fail because we are having some issue with that specific job

**Pull request merged**